packages/material-ui-system/src/createTheme/createTheme.d.ts

```diff
@@ -0,0 +1,40 @@
+import { Breakpoints, BreakpointsOptions } from './createBreakpoints';
+import { Mixins, MixinsOptions } from './createMixins';
+import { Shadows } from './shadows';
+import { Shape, ShapeOptions } from './shape';
+import { Spacing, SpacingOptions } from './createSpacing';
+import { Transitions, TransitionsOptions } from './createTransitions';
+
+export { Breakpoint, BreakpointOverrides } from './createBreakpoints';
+
+export type Direction = 'ltr' | 'rtl';
+
+export interface ThemeOptions {
+  shape?: ShapeOptions;
+  breakpoints?: BreakpointsOptions;
+  direction?: Direction;
+  mixins?: MixinsOptions;
+  palette?: Record<string, any>;
+  shadows?: Shadows;
+  spacing?: SpacingOptions;
+  transitions?: TransitionsOptions;
+}
```

Comment:
This is the important part to focus on when reviewing. Is this structure enough for a baseline theme? Should we include here the components, typography, zIndex although we don't provide any defaults?

Maybe something like:
```suggestion
export interface ThemeOptions {
  shape?: ShapeOptions;
  breakpoints?: BreakpointsOptions;
  direction?: Direction;
  mixins?: MixinsOptions;
  palette?: Record<string, any>;
  shadows?: Shadows;
  spacing?: SpacingOptions;
  transitions?: TransitionsOptions;
  components?: Record<string, any>;
  zIndex?: Record<string, number>;
  typography?: Record<string, TypographyStyleOptions> & FontStyleOptions;
  
  
}
```

Reply:
Ok, let me add these 👍

Reply:
Extended, I needed to add them as optional in the `Theme` interface and used `any`, as we are not implementing anything related to them in the system.